Sunrise/ViewControllers/ScannerViewController.swift

```diff
@@ -0,0 +1,158 @@
+
+        previewLayer.frame = view.layer.bounds
+        previewLayer.videoGravity = AVLayerVideoGravityResizeAspectFill
+        view.layer.addSublayer(previewLayer)
+
+        viewModel?.isCapturing.value = true
+    }
+
+    /**
+        Method used to present errors related to capture device capabilities and permissions.
+    */
+    private func presentCaptureError() {
+        let authorizationStatus = AVCaptureDevice.authorizationStatusForMediaType(AVMediaTypeVideo)
+
+        let alertController = UIAlertController(
+                title: viewModel?.errorTitle,
+                message: authorizationStatus == .Denied ? viewModel?.permissionError : viewModel?.capabilitiesError,
+                preferredStyle: .Alert
+                )
+        if authorizationStatus == .Denied {
+            alertController.addAction(UIAlertAction(title: "Settings", style: .Cancel, handler: { _ in
+                if let appSettingsURL = NSURL(string: UIApplicationOpenSettingsURLString) {
+                    UIApplication.sharedApplication().openURL(appSettingsURL)
+                }
+                self.navigationController?.popViewControllerAnimated(true)
+            }))
+        }
+        alertController.addAction(UIAlertAction(title: "OK", style: .Default, handler: { _ in
+            self.navigationController?.popViewControllerAnimated(true)
+        }))
+
+        presentViewController(alertController, animated: true, completion: nil)
+    }
+
+    // MARK: - Bindings
+
+    private func bindViewModel() {
+        guard let viewModel = viewModel where isViewLoaded() else { return }
+
+        viewModel.isLoading.producer
+        .observeOn(UIScheduler())
+        .startWithNext({ isLoading in
+            if isLoading {
+                SVProgressHUD.show()
+            } else {
+                SVProgressHUD.dismiss()
+            }
+        })
+
+        viewModel.isCapturing.producer
+        .observeOn(UIScheduler())
+        .startWithNext({ [weak self] isCapturing in
+            if isCapturing {
+                self?.captureSession.startRunning()
+            } else {
+                self?.captureSession.stopRunning()
+            }
+        })
+
+        viewModel.scannedProduct.producer
+        .observeOn(UIScheduler())
+        .startWithNext({ [weak self] scannedProduct in
+            if scannedProduct != nil {
+                self?.performSegueWithIdentifier("showScannedProduct", sender: self)
+            }
+        })
+
+        observeAlertMessageSignal(viewModel: viewModel)
+    }
+
+    // MARK: - Navigation
+
+    override func prepareForSegue(segue: UIStoryboardSegue, sender: AnyObject?) {
+        if let productViewController = segue.destinationViewController as? ProductViewController, viewModel = viewModel,
+                product = viewModel.scannedProduct.value {
+            let productDetailsViewModel = ProductViewModel(product: product)
+            productViewController.viewModel = productDetailsViewModel
```

Comment:
I saw that the model already contains `isMatchingVariant`, but are you using it?

In this case it would be quite nice to select the correct variant. E.g. if you scan a shirt in size `L`, that variant would automatically be selected. (I don't know if we have a test case for this yet...)